### PR TITLE
docs: parser implementation guide

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

> [!IMPORTANT]
> This guide is incomplete. The chapters around lexer, token source and parsing require more work and examples. Hopefully, we can add more documentation and examples in later PRs.

### Changes
- Creates a new `CONTRIBUTING.md` file inside `biome_parser` create.
- In this document, I unified what we had in the `lib.rs` file and what we had in the `README.md` of the codegen. Now it's all inside one single document.
- I removed references to `cargo codegen grammar` and created a `just` script, which is easier to use:
	- `just gen-grammar` for all grammars`
	- `just gen-grammar html` for just HTML grammar 

The idea is to sunset `cargo/config.toml` and move everything under `justfile`.

## Test Plan

Proof read

### CODE REVIEW

File: crates/biome_parser/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+  //   (A)                   -- same as A
+  //   label:A               -- suggested name for field of AST node
+  ```
+1. Create a new file called `src/html_kinds_src.rs`. This file must return a static `KindSrc`.
+1. Create two new crates: `biome_html_syntax` and `biome_html_factory`. Use `cargo new --lib crates/biome_html_syntax`.
+1. Create a `generated/` folder inside the `src/` folder of the newly created crates.
+1. Add a new variant to `LanguageKind`, inside `language_kind.rs` file. The new variant will be `Html`. You'll have to implement
+   all methods and cover the new variant.
+1. Add a new prefix `html_` to `LANGUAGE_PREFIXES` inside `language_kind.rs`.
+1. Once you covered all variants, run the command `cargo codegen grammar`.
```
Comment (Sec-ant) [inline crates/biome_parser/CONTRIBUTING.md:46]: list item number should be increasing

Comment (ematipico) [reply crates/biome_parser/CONTRIBUTING.md:46]: That's the trick, if you leave them to all `1` they will be automatically **rendered** with increased numbers :) 

This means that if you need to add a new item to the list, you don't need to update all the numbers. Check the [rendered document](https://github.com/biomejs/biome/blob/docs/better-parser-contribution-guide/crates/biome_parser/CONTRIBUTING.md#create-a-new-language)

Comment (Sec-ant) [reply crates/biome_parser/CONTRIBUTING.md:46]: 🤯 I just learned new tricks.

File: crates/biome_parser/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+From the root of this repository, head to `xtask/codegen` folder. That's where you'll start doing your work.
+
+Let's say you want to create a new language that has the extension `html`, you'll have to follow these instructions:
+
+1. Create a new `html.ungram` file in this folder.
+   Add this legend to the `.ungram` file.
+
+  ```
+  // This grammar specifies the structure of Rust's concrete syntax tree.
```
Comment (nhedger) [inline crates/biome_parser/CONTRIBUTING.md:22]:
```suggestion
  // This grammar specifies the structure of [LANGUAGE]'s concrete syntax tree.
```

File: crates/biome_parser/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+advance in the parsing process and consume tokens - if it returns `Absent`.
+
+
+#### Parse children
+The parse rules will guide you in how to write your implementation and the parser infrastructure provides the following convenience APIs:
+
+* Optional token `'ident'?`: Use `p.eat(token)`. It eats the next token if it matches the passed-in token.
+* Required token `'ident'`: Use`p.expect(token)`. It eats the next token if it matches the passed-in token.
+It adds an `Expected 'x' but found 'y' instead` error and a missing marker if the token isn't present in the source code.
+* Optional node `body: JsBlockStatement?`: Use`parse_block_statement(p).or_missing(p)`. It parses the block if it is present in the source code and adds a missing marker if it isn't.
```
Comment (arendjr) [inline crates/biome_parser/CONTRIBUTING.md:217]: Question: When is `or_missing()` actually necessary? I’ve found missing markers for optional nodes to be added automatically so far.

File: crates/biome_parser/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+};
+```
+
+Let's run through this step by step:
+
+```rust, ignore
+parsed_element.or_recover(
+    p,
+    &ParseRecoveryTokenSet::new(JS_BOGUS_STATEMENT, STMT_RECOVERY_SET),
+    js_parse_error::expected_case,
```
Comment (arendjr) [inline crates/biome_parser/CONTRIBUTING.md:301]: The docs say we’re parsing an array, but the example seems to be for switch statements?

**Review by arendjr — APPROVED** [review]
Nice! Just one example seems to be off?

File: crates/biome_parser/CONTRIBUTING.md (hunk 3 above)
Comment (ematipico) [reply crates/biome_parser/CONTRIBUTING.md:217]: Docs are **outdated** :D 

I just copied the content form `lib.rs`. Happy to review it later

File: crates/biome_parser/CONTRIBUTING.md (hunk 4 above)
Comment (ematipico) [reply crates/biome_parser/CONTRIBUTING.md:301]: Definitely outdate, I will take care of it later

File: crates/biome_parser/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+
+ Present(m.complete(p, JS_IF_STATEMENT));
+}
+```
+
+Hold on, what are these *missing* markers? Biome's AST facade uses fixed offsets to retrieve a particular child from a node.
+For example, the 3rd child of the if statement is the condition. However, the condition would become the second element
+if the opening parentheses `(` isn't present in the source text. That's where missing elements come into play.
+
+### Parsing Lists & Error Recovery
```
[the remark this answers is not in the document]
Comment (ematipico) [reply crates/biome_parser/CONTRIBUTING.md:247]: Thank you. I will follow up with another PR :)

**Pull request merged**